api/src/main/java/org/pmiops/workbench/disks/DiskService.java

```diff
@@ -0,0 +1,38 @@
+package org.pmiops.workbench.disks;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.pmiops.workbench.leonardo.LeonardoApiClient;
+import org.pmiops.workbench.leonardo.model.LeonardoListPersistentDiskResponse;
+import org.pmiops.workbench.model.Disk;
+import org.pmiops.workbench.utils.mappers.LeonardoMapper;
+import org.pmiops.workbench.workspaces.WorkspaceService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class DiskService {
+  private final LeonardoMapper leonardoMapper;
+  private final LeonardoApiClient leonardoNotebooksClient;
+  private final WorkspaceService workspaceService;
+
+  @Autowired
+  public DiskService(
+      LeonardoMapper leonardoMapper,
+      LeonardoApiClient leonardoNotebooksClient,
+      WorkspaceService workspaceService) {
+    this.leonardoMapper = leonardoMapper;
+    this.leonardoNotebooksClient = leonardoNotebooksClient;
+    this.workspaceService = workspaceService;
+  }
+
+  public List<Disk> findByWorkspaceNamespace(String workspaceNamespace) {
```

Comment:
1) Seems like we are on the same page here.

2) I would agree that all non-Leo related functionality should be pulled out of the Leo service. I think that that could be handled in a follow up PR and hopefully after we have built some consensus as a team about the benefits of this pattern (meeting tomorrow).

3) The way that I would envision this would be that in addition to taking disk functionality out of the Leo service, that any disk functionality in the app and runtime services would be done by calling functions from within the disk service.

Reply:
Thanks for response! Without 2 and 3 addressed, introducing the new DiskService that does so little things add extra confusion and complexity. Fine of having this services for now then address 2/3 asap. Otherwise, we should consider delete this service and just move workspace look up into Controller. 
(1: Workspace Lookup seems to be a generic conversion for most RW API calls, then it make sense to have this in Controller level then support inject Workspace like we did for `User` in controllers.
2: The workspace lookup in Controller is already universal solution for all other calls,  introducing a new pattern without planing to refactor others (need to agree with 1 first), might makes the two pattern exists in our codebase forever...
 )

Reply:
Yeah, good call. I think that points 2 and 3 should be addressed asap. We should talk about this during the meeting tomorrow, and if we cannot agree on an approach, we can consider deleting the service.